controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -757,4 +767,102 @@ export class FederatedGraphRepository {
         }) as GraphApiKeyDTO,
     );
   }
+
+  public async createGraphCryptoKeyPairs(input: {
+    organizationId: string;
+    federatedGraphId: string;
+  }): Promise<RouterRequestKeysDTO> {
+    const keys = await generateKeyPair('ES256');
+
+    const privateKey = (keys.privateKey as KeyObject).export({
+      format: 'pem',
+      type: 'pkcs8',
+    });
+    const publicKey = (keys.publicKey as KeyObject).export({
+      format: 'pem',
+      type: 'spki',
+    });
+
+    const items = await this.db
+      .insert(graphRequestKeys)
+      .values({
+        privateKey: privateKey.toString(),
```

Comment:
That wouldn't be much safer. In cloud we ensure that all data is encrypted at rest. You would need system access to read the data. If the attacker would get access to the process he can also read the environment variable and because it is OSS he also knows how to decrypt it. We can't salt or hash the private key because we need the orginal content. Does it makes sense?